File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums):
        
        a = len(nums)
        i = 0
        while i < a:
            j = i + 1
            while j < a:
                if nums[i] == nums[j]:
                    nums.pop(j)
                    a -= 1  # update length after popping
                else:
                    j += 1  # only increment if no pop
            i += 1
        return len(nums)

    

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
import java.util.Arrays;

class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;

        int j = 0; // position for placing the next unique element
        for (int i = 1; i < nums.length; i++) {
            if (nums[j] != nums[i]) {
                j++;
                nums[j] = nums[i];
                // j++;
            }
        }

        return j+1; // number of unique elements
    }
}
